x-pack/plugins/cloud_security_posture/server/lib/check_index_status.ts

```diff
@@ -11,14 +11,28 @@ import { IndexStatus } from '../../common/types';
 export const checkIndexStatus = async (
   esClient: ElasticsearchClient,
   index: string,
-  logger: Logger
+  logger: Logger,
+  postureType: 'cspm' | 'kspm' | 'all' = 'all'
 ): Promise<IndexStatus> => {
+  const query =
+    postureType === 'all'
+      ? {
+          match_all: {},
+        }
+      : {
+          bool: {
+            filter: {
+              term: {
+                'rule.benchmark.posture_type': postureType,
```

Comment:
since `rule.benchmark.posture_type` is a new field, we need to add backward computability here.